### PR TITLE
add libcorrect/20181010

### PR DESCRIPTION
**Pull request by btashton**
Specify library name and version:  **libcorrect/20181010**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by btashton** [comment]
Something is broken with the test executable generated with windows shared builds, static seem to be fine.

**Comment by danimtb** [comment]
Does the library support shared build on Windows? If not, we can just add ConanInvalidConfiguration for it

File: recipes/libcorrect/all/patches/0001-Support-CMake-BUILD_SHARED_LIBS.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ endif()
+ 
+ add_subdirectory(util)
+@@ -85,18 +90,16 @@ add_subdirectory(tests)
+ add_subdirectory(tools)
+ # add_subdirectory(benchmarks)
+ 
+-install(TARGETS correct correct_static
++install(TARGETS correct
+         DESTINATION lib)
```
Comment (madebr) [inline recipes/libcorrect/all/patches/0001-Support-CMake-BUILD_SHARED_LIBS.patch]:
```suggestion
+        ARCHIVE DESTINATION lib LIBRARY DESTINATION lib RUNTIME DESTINATION bin)
```

File: recipes/libcorrect/all/patches/0001-Support-CMake-BUILD_SHARED_LIBS.patch [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+-set_target_properties(fec_shim_shared PROPERTIES OUTPUT_NAME "fec")
++add_library(fec_shim ${LIB_TYPE} EXCLUDE_FROM_ALL src/fec_shim.c ${correct_obj_files})
++set_target_properties(fec_shim PROPERTIES OUTPUT_NAME "fec")
+ add_custom_target(fec-shim-h COMMAND ${CMAKE_COMMAND} -E copy ${PROJECT_SOURCE_DIR}/include/fec_shim.h ${PROJECT_BINARY_DIR}/include/fec.h)
+-add_custom_target(shim DEPENDS fec_shim_static fec_shim_shared fec-shim-h)
++add_custom_target(shim DEPENDS fec_shim fec-shim-h)
+ 
+-install(TARGETS fec_shim_static fec_shim_shared
++install(TARGETS fec_shim
+         DESTINATION lib
```
Comment (madebr) [inline recipes/libcorrect/all/patches/0001-Support-CMake-BUILD_SHARED_LIBS.patch]:
```suggestion
+        ARCHIVE DESTINATION lib LIBRARY DESTINATION lib RUNTIME DESTINATION bin
```

File: recipes/libcorrect/all/patches/0002-Export-symbols-for-windows.patch [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 6169ac4..6bc863d 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -5,6 +5,7 @@ include(CheckIncludeFiles)
+ include(CheckCSourceCompiles)
+ include(CMakePushCheckState)
+ include(CheckCCompilerFlag)
++include(GenerateExportHeader)
```
Comment (madebr) [inline recipes/libcorrect/all/patches/0002-Export-symbols-for-windows.patch]: I will steal this information and use it for future projects!
Thanks for the tip!

Comment (uilianries) [reply recipes/libcorrect/all/patches/0002-Export-symbols-for-windows.patch]: I didn't know about too!

File: recipes/libcorrect/all/patches/0001-Support-CMake-BUILD_SHARED_LIBS.patch (hunk 2 above)
Comment (btashton) [reply recipes/libcorrect/all/patches/0001-Support-CMake-BUILD_SHARED_LIBS.patch]: Thanks for this.  Had to make a slight modification, but it pointed me in the right direction.

**Comment by danimtb** [comment]
@btashton are you done with the PR? Can we merge it?

**Comment by btashton** [comment]
Should be good. Using it as a dependency in another package.

**Pull request merged**